controllers/metal3.io/host_state_machine.go

```diff
@@ -254,7 +254,7 @@ func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result acti
 	// provisioner and take no further action
 	// Note this doesn't change the current state, only the OperationalStatus
 	if hasDetachedAnnotation(hsm.Host) {
-		// Only allow detaching hosts in Provisioned/ExternallyProvisioned states
+		// Only allow detaching hosts in Provisioned/ExternallyProvisioned/Ready states
 		switch info.host.Status.Provisioning.State {
 		case metal3v1alpha1.StateProvisioned, metal3v1alpha1.StateExternallyProvisioned, metal3v1alpha1.StateReady:
```

Comment:
Since we're here, we should probably include StateAvailable wherever we mention StateReady, although it feels like progress on #340 has stalled.

Reply:
Included `StateAvailable` everywhere as suggested by @zaneb, thank you. 
/cc @hardys